Publish Task additional arguments not honored

I've been trying to debug an issue where tfx is failing to validate our extension during publishing.  The task is failing with this semi-cryptic error:
`2017-12-20T19:16:57.8041612Z error: Error: Validation timed out. There may be a problem validating your extension. Please try again later.`
`2017-12-20T19:16:57.8261928Z ##[error]tfx failed with error: Error: d:\a\_tools\tfx.cmd failed with return code: 4294967295`

Looking at the task.json, I see:
`{   "name": "arguments",
      "type": "string",
      "label": "Arguments",
      "defaultValue": "",
      "helpMarkDown": "Additional arguments passed to the package and publishing tool.",
      "required": false,
      "groupName": "advanced"
    }`
However, I could not find where these arguments were processed or added to the tfx command.

In order to debug further, I added '--trace-level debug' to the 'Arguments' under the Advanced variable group.  However, I did not see any additional debug output.

Side note - looking at the tfx-cli history, I believe the underlying issue is related to a change to add faster validation values for non-public releases.

--- a/BuildTasks/Common/Common.ts
+++ b/BuildTasks/Common/Common.ts
@@ -137,6 +137,18 @@ export function validateAndSetTfxManifestArguments(tfx: ToolRunner): (() => void
         jsonOverrides.version = extensionVersion;
     }
 
+    const noWaitValidation = tl.getBoolInput("noWaitValidation", false);
+    if (noWaitValidation) {
+        tl.debug(`Not waiting for validation.`);
+        tfx.arg("--no-wait-validation");
+    }
+
+    const bypassLocalValidation = tl.getBoolInput("bypassLocalValidation", false);
+    if (bypassLocalValidation) {
+        tl.debug(`Bypassing local validation.`);
+        tfx.arg("--bypass-validation");
+    }
+
     let overrideFilePath: string;
     if (jsonOverrides) {
         // Generate a temp file
@@ -146,7 +158,11 @@ export function validateAndSetTfxManifestArguments(tfx: ToolRunner): (() => void
         tfx.arg(["--overrides-file", overrideFilePath]);
     }
 
-    tfx.line(tl.getInput("arguments", false));
+    const args = tl.getInput("arguments", false);
+    if (args) {
+        tl.debug(`Adding additional arguments: ${args}.`);
+        tfx.line(args);
+    }
 
     return () => deleteBuildTempFile(overrideFilePath);
 }

--- a/BuildTasks/Common/Common.ts
+++ b/BuildTasks/Common/Common.ts
@@ -227,7 +227,7 @@ export function runTfx(cmd: (tfx: ToolRunner) => void) {
     tl.mkdirP(path.join(agentToolsPath, "/node_modules/"));
 
     const npm = new trl.ToolRunner(tl.which("npm", true));
-    npm.arg(["install", "tfx-cli@0.4.8", "--prefix", agentToolsPath]);
+    npm.arg(["install", "tfx-cli", "--prefix", agentToolsPath]);
 
     npm.exec().then(code => {
         tfx = new trl.ToolRunner(tl.which(tfxLocalPath) || tl.which(tfxLocalPathBin, true));

--- a/BuildTasks/ExtensionVersion/task.json
+++ b/BuildTasks/ExtensionVersion/task.json
@@ -9,7 +9,7 @@
   "version": {
     "Major": 1,
     "Minor": 1,
-    "Patch": 44
+    "Patch": 47
   },
   "visibility": [
     "Build",

--- a/BuildTasks/InstallExtension/task.json
+++ b/BuildTasks/InstallExtension/task.json
@@ -9,7 +9,7 @@
   "version": {
     "Major": 1,
     "Minor": 1,
-    "Patch": 44
+    "Patch": 47
   },
   "visibility": [
     "Build",

--- a/BuildTasks/PackageExtension/task.json
+++ b/BuildTasks/PackageExtension/task.json
@@ -9,7 +9,7 @@
   "version": {
     "Major": 1,
     "Minor": 1,
-    "Patch": 44
+    "Patch": 47
   },
   "visibility": [
     "Build",

--- a/BuildTasks/PackageExtension/task.json
+++ b/BuildTasks/PackageExtension/task.json
@@ -173,6 +173,15 @@
       }
     },
     {
+      "name": "bypassLocalValidation",
+      "type": "boolean",
+      "label": "Bypass local validation",
+      "defaultValue": "false",
+      "helpMarkDown": "Bypass local validation.",
+      "required": false,
+      "groupName": "advanced"
+    },
+    {
       "name": "arguments",
       "type": "string",
       "label": "Arguments",

--- a/BuildTasks/PublishExtension/PublishExtension.ts
+++ b/BuildTasks/PublishExtension/PublishExtension.ts
@@ -113,6 +113,24 @@ common.runTfx(tfx => {
         }
     }
 
+    const noWaitValidation = tl.getBoolInput("noWaitValidation", false);
+    if (noWaitValidation) {
+        tl.debug(`Not waiting for validation.`);
+        tfx.arg("--no-wait-validation");
+    }
+
+    const bypassLocalValidation = tl.getBoolInput("bypassLocalValidation", false);
+    if (bypassLocalValidation) {
+        tl.debug(`Bypassing local validation.`);
+        tfx.arg("--bypass-validation");
+    }
+
+    const args = tl.getInput("arguments", false);
+    if (args) {
+        tl.debug(`Adding additional arguments: ${args}.`);
+        tfx.line(args);
+    }
+
     runBeforeTfx.then(() => {
         const outputStream = new common.TfxJsonOutputStream(true);
         tfx.exec(<any>{ outStream: outputStream, failOnStdErr: true }).then(code => {

--- a/BuildTasks/PublishExtension/task.json
+++ b/BuildTasks/PublishExtension/task.json
@@ -9,7 +9,7 @@
   "version": {
     "Major": 1,
     "Minor": 1,
-    "Patch": 44
+    "Patch": 47
   },
   "visibility": [
     "Build",

--- a/BuildTasks/PublishExtension/task.json
+++ b/BuildTasks/PublishExtension/task.json
@@ -232,6 +232,24 @@
       "visibleRule": "extensionVisibility = private || extensionVisibility = privatepreview || extensionVisibility = default"
     },
     {
+      "name": "bypassLocalValidation",
+      "type": "boolean",
+      "label": "Bypass local validation",
+      "defaultValue": "false",
+      "helpMarkDown": "Bypass local validation.",
+      "required": false,
+      "groupName": "advanced"
+    },
+    {
+      "name": "noWaitValidation",
+      "type": "boolean",
+      "label": "Don't wait for validation",
+      "defaultValue": "false",
+      "helpMarkDown": "Don't block command for extension validation.",
+      "required": false,
+      "groupName": "advanced"
+    },
+    {
       "name": "arguments",
       "type": "string",
       "label": "Arguments",

--- a/README.md
+++ b/README.md
@@ -26,6 +26,6 @@ This extension provides build and release tasks for packaging and publishing Vis
 If you make a change to one or more of the actual tasks, you need to update the task(s) version in the appropriate task.json file. You can also run `setTaskVersion.js` and pass version to apply to all tasks, for example:
 
 ```
-node scripts/setTaskVersion.js 1.1.44
+node scripts/setTaskVersion.js 1.1.47
 ```
 

--- a/vss-extension.json
+++ b/vss-extension.json
@@ -3,7 +3,7 @@
   "id": "vsts-developer-tools-build-tasks",
   "publisher": "ms-devlabs",
   "name": "CI/CD Tools for VSTS Extensions",
-  "version": "1.1.46",
+  "version": "1.1.47",
   "description": "CI/CD tasks for packaging and publishing VSTS extensions to the Visual Studio Marketplace.",
   "tags": [
     "Extension",

